module-system/sov-state/src/prover_storage.rs

```diff
@@ -46,16 +46,23 @@ impl<S: MerkleProofSpec> ProverStorage<S> {
             .db
             .get_value_option_by_key(self.db.get_next_version(), key.as_ref())
         {
-            Ok(value) => value.map(StorageValue::new_from_bytes),
+            Ok(value) => value.map(Into::into),
             // It is ok to panic here, we assume the db is available and consistent.
             Err(e) => panic!("Unable to read value from db: {e}"),
         }
     }
+
+    fn get_root_hash(&self, version: Version) -> Result<RootHash, anyhow::Error> {
+        let temp_merkle: JellyfishMerkleTree<'_, StateDB, S::Hasher> =
+            JellyfishMerkleTree::new(&self.db);
```

Comment:
Shouldn't an instance of the tree live with the type? Otherwise, shouldn't we delegate this instantiation and type choice to some attribute of `StateDB`?

Reply:
I think this way to instantiate the JellyfishMerkleTree is done in multiple places in the codebase. We can open an issue to change and standardize your approach